src/main/java/chatbot/impl/TaskStorageImpl.java

```diff
@@ -0,0 +1,142 @@
+package chatbot.impl;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import chatbot.Task;
+import chatbot.TaskStorage;
+import chatbot.impl.tasks.AbstractTask;
+
+/**
+ * Implements the TaskStorage interface to manage tasks in a custom file-based storage system.
+ */
+public class TaskStorageImpl implements TaskStorage {
```

Comment:
Same as above, would it be better to have named the interface TaskStorageInterface(or similar) so that the class itself could just be named as TaskStorage?